enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/CypherComparisonSupport.scala

```diff
@@ -95,209 +95,153 @@ trait CypherComparisonSupport extends CypherTestSupport {
-                                                    ignoreScenarios: TestConfiguration,
-                                                    query: String,
-                                                    checkPlans: Boolean,
-                                                    params: (String, Any)*): InternalExecutionResult = {
-    updateWithAndExpectPlansToBeSimilar(expectedSuccessFrom, ignoreScenarios, query, checkPlans, () => {}, params: _*)
+      val tryResult: Try[InternalExecutionResult] = Try(innerExecute(s"CYPHER ${thisScenario.preparserOptions} $query", params.toMap))
+      tryResult match {
+        case Success(result) =>
+          fail("Unexpectedly Succeeded in " + thisScenario.name)
+        case Failure(e: CypherException) =>
+          if (expectedToFailWithSpecificMessage) {
+            if (e.getMessage == null || !e.getMessage.contains(message)) {
+              fail("Correctly failed in " + thisScenario.name + " but instead of '" + message +
+                "' the error message was '" + e.getMessage + "'")
+            }
+          } else {
+            if (e.getMessage.contains(message)) {
+              fail("Unexpectedly (but correctly!) failed in " + thisScenario.name + " with the correct message. Did you forget to add this config?")
+            }
+            // It failed like expected, and we did not specify any message for this config
+          }
+        case Failure(e: Throwable) => throw e
+      }
+    }
   }
 
-  protected def updateWithAndExpectPlansToBeSimilar(expectedSuccessFrom: TestConfiguration,
-                                                    ignoreScenarios: TestConfiguration,
-                                                    query: String,
-                                                    checkPlans: Boolean,
-                                                    executeBefore: () => Unit,
-                                                    params: (String, Any)*): InternalExecutionResult = {
-    assertNoOverlap(expectedSuccessFrom, ignoreScenarios)
-    val runAgainstScenarios = Configs.AbsolutelyAll - ignoreScenarios
-
-    val firstScenario = extractFirstScenario(expectedSuccessFrom)
+  protected def executeWith(expectedSuccessFrom: TestConfiguration,
+                            query: String,
+                            ignoreResults: TestConfiguration = Configs.Empty,
+                            ignorePlans: TestConfiguration = Configs.AllRulePlanners,
+                            executeBefore: () => Unit = () => {},
+                            params: Map[String, Any] = Map.empty): InternalExecutionResult = {
+    val compareResults = expectedSuccessFrom -  ignoreResults
+    val comparePlans = expectedSuccessFrom - ignorePlans
+    val firstScenario = extractFirstScenario(expectedSuccessFrom, compareResults, comparePlans)
+
+    // let this throw to enable catching exceptions outside
+    firstScenario.prepare()
```

Comment:
I commented this block out and ran all acceptance and all compatibility tests. No failures. Can you point me to the test(s) that require this to be here?

Reply:
`test("should refuse to execute index seeks using inequalities over different types")` in `NodeIndexSeekByRangeAcceptanceTest` is reliant on this being here (when it has been reverted to actually check for the exception again)